h/api/search/query.py

```diff
@@ -97,29 +96,24 @@ def build(request_params, userid=None, search_normalized_uris=False):
     }
 
 
-def _match_clause_for_uri(uristr):
+def _filter_for_uri_match(uristr):
     """Return an Elasticsearch match clause dict for the given URI."""
     uristrs = uri.expand(uristr)
-    matchers = [{"match": {"uri": u}} for u in uristrs]
+    clauses = [{"match": {"uri": u}} for u in uristrs]
 
-    if len(matchers) == 1:
-        return matchers[0]
-    return {
-        "bool": {
-            "minimum_should_match": 1,
-            "should": matchers
-        }
-    }
+    if len(clauses) == 1:
+        return {"query": clauses[0]}
+    return {"query": {"bool": {"should": clauses}}}
```

Comment:
Couldn't this just be a terms filter? `{"uri": uristrs}` <https://www.elastic.co/guide/en/elasticsearch/guide/current/_most_important_queries_and_filters.html#_terms_filter> (and in the cases where these only one uri, return a `term` filter)